### PR TITLE
Unit tests for training job

### PR DESCRIPTION
**Pull request by ryansteakley**
Description of changes:

- Sdk.go coverage 70.1%

- Custom code coverage > 80%

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/resource/training_job/testdata/test_suite.yaml [hunk 1]
```diff
@@ -1,17 +1,193 @@
...
+      - name: "Delete=Successful"
+        description: "This test checks if the Training job is deleted successfully"
+        given:
+          desired_state: "v1alpha1/readone/desired/after_create_inprogress.yaml"
+          svc_api:
+            - operation: StopTrainingJobWithContext
+            - operation: DescribeTrainingJobWithContext
+              error:
+                code: ValidationException
+                message: "Requested resource not found"
```
Comment (akartsky) [inline pkg/resource/training_job/testdata/test_suite.yaml:172]: Describe would still find the resource as there is no delete function in TrainingJob  
Our controller should work without the error response

File: pkg/resource/training_job/testdata/test_suite.yaml [hunk 2]
```diff
@@ -1,17 +1,193 @@
...
+          error: nil      
+      - name: "Delete=MarkUnmanaged"
+        description: "This test checks if the Training job returns nil to mark the resource Unmanaged if job not in InProgress state"
+        given:
+          desired_state: "v1alpha1/readone/observed/completed.yaml"
+          svc_api:
+            - operation: StopTrainingJobWithContext
+        invoke: Delete
+        expect:
+          error: nil
```
Comment (akartsky) [inline pkg/resource/training_job/testdata/test_suite.yaml:184]: what are we testing here ?

File: pkg/resource/training_job/testdata/v1alpha1/readone/desired/stopping.yaml [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    value: xgboost
+  - key: environment
+    value: testing
+  - key: customer
+    value: test-user
+  trainingJobName: xgboost-training-job
+status:
+  ackResourceMetadata:
+    arn: arn:aws:sagemaker:us-west-2:123456789012:training-job/xgboost-training-job
+    trainingJobStatus: Completed
```
Comment (akartsky) [inline pkg/resource/training_job/testdata/v1alpha1/readone/desired/stopping.yaml:56]: This should be "Stopped" or "Stopping" right ?

File: pkg/resource/training_job/testdata/v1alpha1/readone/observed/stopping.yaml [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+  ackResourceMetadata:
+    arn: arn:aws:sagemaker:us-west-2:123456789012:training-job/xgboost-training-job
+    ownerAccountID: ""
+  conditions:
+  - lastTransitionTime: "0001-01-01T00:00:00Z"
+    message: TrainingJob is in Stopping status.
+    status: "False"
+    type: ACK.ResourceSynced
+  secondaryStatus: Starting
+  trainingJobStatus: Stopping
```
Comment (akartsky) [inline pkg/resource/training_job/testdata/v1alpha1/readone/observed/stopping.yaml:107]: How did this unit test pass considering we have "Completed" in desired state and "Stopping" in observed state ?

File: pkg/resource/training_job/testdata/sdkapi/describe/stopping_describe.json [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+    "ProfilerRuleEvaluationStatuses": null,
+    "ProfilingStatus": "Disabled",
+    "ResourceConfig": {
+        "InstanceCount": 1,
+        "InstanceType": "ml.m4.xlarge",
+        "VolumeKmsKeyId": null,
+        "VolumeSizeInGB": 5
+    },
+    "RoleArn": "arn:aws:iam::123456789012:role/service-role/AmazonSageMaker",
+    "SecondaryStatus": "Starting",
```
Comment (mbaijal) [inline pkg/resource/training_job/testdata/sdkapi/describe/stopping_describe.json:160]: is this correct ?

File: pkg/resource/training_job/testdata/sdkapi/describe/success_describe_debugger_variation.json [hunk 6]
```diff
@@ -0,0 +1,262 @@
...
+    ],
+    "StoppingCondition": {
+        "MaxRuntimeInSeconds": 86400,
+        "MaxWaitTimeInSeconds": null
+    },
+    "TensorBoardOutputConfig": null,
+    "TrainingEndTime": null,
+    "TrainingJobArn": "arn:aws:sagemaker:us-west-2:123456789012:training-job/xgboost-training-debugger-job",
+    "TrainingJobName": "xgboost-training-debugger-job",
+    "TrainingJobStatus": "InProgress",
```
Comment (mbaijal) [inline pkg/resource/training_job/testdata/sdkapi/describe/success_describe_debugger_variation.json:257]: same comment on filename - ideally, inprogress != success

File: pkg/resource/training_job/testdata/v1alpha1/readone/desired/stopping.yaml (hunk 3 above)
Comment (mbaijal) [reply pkg/resource/training_job/testdata/v1alpha1/readone/desired/stopping.yaml:56]: @ryansteakley I am also a little confused about these job statuses, secondary statuses and debugger status - could just review these one more time and make sure they are as expected ?

**Review by akartsky — APPROVED** [review]

**Pull request merged**